inst/tests/tests.Rraw

```diff
@@ -18612,3 +18612,49 @@ test(2263.3, options=list(datatable.verbose=TRUE, datatable.optimize=0L), names(
 test(2263.4, options=list(datatable.verbose=TRUE, datatable.optimize=Inf), dt[, .N, b], data.table(b=dt$b, N=1L), output="GForce optimized j to")
 test(2263.5, options=list(datatable.verbose=TRUE, datatable.optimize=Inf), dt[, .N, .(b,c)], data.table(b=dt$b, c=dt$c, N=1L), output="GForce optimized j to")
 test(2263.6, options=list(datatable.verbose=TRUE, datatable.optimize=Inf), names(attributes(dt[, .N, b]$b)), c("class", "att"), output="GForce optimized j to")
+
+# tests for printing indices alongside data.tables
+NN = 200
+set.seed(2024)
+DT = data.table(
+ grp1 = sample(100, NN, TRUE),
+ grp2 = sample(90, NN, TRUE),
+ grp3 = sample(80, NN, TRUE)
+)
+setkey(DT, grp1, grp2)
+setindex(DT, grp1, grp3)
+dt2 = copy(DT)
+ans = c(
+ "     grp1 grp2 grp3 index__grp1__grp3",
```

Comment:
I wonder if it's possibly beneficial to use a different separator for the `index` stub vs. the columns themselves? E.g. we could do `index__grp1_grp3` or `index_grp1__grp3`. For the former, we'd do `paste0("index__", vapply_1c(indices(x, vectors=TRUE), paste, collapse="_"))`. Uglier but not awful.

Reply:
I like the look of the former better, but I think it would be beneficial for our column names to accurately reflect the output of `indices(x)`, like `'grp1__grp3'`. This also makes sense as the existing printing of indices show indices like: `Indices: <grp1__grp3>`

Reply:
Since it's a pseudo-column, we might want to do more to emphasize that the index can't actually be accessed as a normal column, maybe making the name non-syntactic is enough (and also helps w the issue raised here):

`index:grp1__grp3` or even `` `index:grp1__grp3` ``

Reply:
Yes this looks great